### PR TITLE
fix: Update line-splitting logic in batched CSV reader

### PR DESCRIPTION
**Pull request by nameexhaustion**
Updates to use the new `CountLines` instead of the previous `accept_line` based logic, which had some incorrect edge cases.

### CODE REVIEW

File: crates/polars-lazy/src/tests/streaming.rs [hunk 1]
```diff
@@ -63,10 +63,7 @@ fn test_streaming_csv() -> PolarsResult<()> {
 
 #[test]
 fn test_streaming_glob() -> PolarsResult<()> {
-    let q = get_csv_glob();
-    let q = q.sort(["sugars_g"], Default::default());
```
Comment (nameexhaustion) [inline crates/polars-lazy/src/tests/streaming.rs:67]: I'm not sure how this was passing before, but there was sort instability for this query

File: py-polars/tests/unit/io/test_csv.py [hunk 2]
```diff
@@ -2129,7 +2131,7 @@ def test_read_csv_only_loads_selected_columns(
             break
         result += next_batch
     del result
-    assert 8_000_000 < memory_usage_without_pyarrow.get_peak() < 13_000_000
+    assert 8_000_000 < memory_usage_without_pyarrow.get_peak() < 20_000_000
```
Comment (nameexhaustion) [inline py-polars/tests/unit/io/test_csv.py:2134]: We use up to a 16MB chunk size now

File: crates/polars-io/src/csv/read/read_impl/batched.rs [hunk 3]
```diff
@@ -76,27 +92,12 @@ impl Iterator for ChunkOffsetIter<'_> {
...
-                    bytes_first_row + 2
-                } else {
-                    1
-                };
                 get_file_chunks_iterator(
                     &mut self.offsets,
                     &mut self.last_offset,
                     self.n_chunks,
-                    self.rows_per_batch * bytes_first_row,
+                    &mut self.chunk_size,
```
Comment (nameexhaustion) [inline crates/polars-io/src/csv/read/read_impl/batched.rs:99]: I've currently changed the (initial) chunk size to the default value copied from `read_impl` from the in-memory CSV reader, and ignoring the user-provided `batch_size` option.

**Comment by nameexhaustion** [comment]
I think the Codspeed report shows that we are executing more instructions - from local testing the wall times show a slight improvement - 

Before - `python .env/data/x.py 2> /dev/null > /dev/null  6.03s user 0.63s system 600% cpu 1.109 total`
After - `python .env/data/x.py 2> /dev/null > /dev/null  6.01s user 0.59s system 603% cpu 1.092 total`

File: crates/polars-io/src/csv/read/read_impl/batched.rs (hunk 3 above)
Comment (ritchie46) [reply crates/polars-io/src/csv/read/read_impl/batched.rs:99]: Yes, good one. I benchmarked that one. I think that one is a good cache friendly default.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**